web-server-lib/web-server/private/dispatch-server-sig.rkt

```diff
@@ -22,13 +23,15 @@
     [listen-ip (or/c string? false/c)]
     [max-waiting exact-nonnegative-integer?]
     [initial-connection-timeout integer?]
+    [response-timeout exact-positive-integer?]
+    [response-send-timeout exact-positive-integer?]
```

Comment:
@Bogdanp IIUC, adding `response-timeout` and `response-send-timeout` elements to `dispatch-server-config^` (and likewise for `web-config^`) would be a breaking change for any programs that define their own units that export `dispatch-server-config^`. I think instead we need to declare a new signature using the `safety-limits` struct. I'm still working out the least-ugly way to do that.

Reply:
I think a fairly elegant solution is to make `dispatch-server-config^` extend a `dispatch-server-config*^` and use `define-values-for-export` to generate the `safety-limits` element definition.

Reply:
Sounds good to me. I think it would be good to add a test specifically for this to prevent these types of breakages in the future. If you don't think you'll get around to it, then let me know and I'll give it a try. 

I had never dealt with units before and it never occurred to me that this is a breaking change  even though I obviously had to change `web-server-unit.rkt` in response to this change so a test would've helped for sure.

Reply:
@Bogdanp What kind of test do you have in mind for this? It seems hard to protect against someone making a breaking change to a signature or unit (probably unintentionally: units are somewhat obscure in modern Racket) and also changing the test suite at the same time—but maybe you're thinking of something I'm not. In any case, probably some comments would be helpful …

Reply:
I was thinking of constructing a dummy `web-server-unit` in a `test-case`.  The test itself doesn't have to assert anything, it just has to fail to compile when the sigs change and have a nice message like `"should not break backwards-compatibility of signatures"`. I realize this is kind of redundant, given that `web-server-unit.rkt` itself would fail given a change to either sig, like it did when I made the change, but this would make it harder to unintentionally make a breaking change.

Thought process after modifying a sig: "ah, of course, I need to also change the unit!"
Thought process after modifying a sig with this "test" in place: "oh crap, this is a breaking change, I'd better ask for advice/rethink my approach!"

Reply:
@Bogdanp Yes, that sounds like a great idea!